### PR TITLE
[Python] Add uniform `create` methods to the rest of the Comb ops.

### PR DESCRIPTION
**Pull request by mikeurbach**
This adds a `create` classmethod to the variadic and other Comb ops,
rounding out the Python bindings sugar for the Comb dialect.

Additionally, this removes the kwargs from the non-variadic Comb ops'
`create` classmethods. This presents the same API as the variadic ops
for the common case where all the operands are known at the time the
op is constructed.

Under the hood, non-variadic ops still use the NamedValueOpView to
continue supporting the less common case of constructing the operands
incrementally. Incremental construction is simply not possible for the
variadic ops, since we need to know how many operands to build at
construction time.

### CODE REVIEW

File: integration_test/Bindings/Python/dialects/comb.py [hunk 1]
```diff
@@ -19,140 +20,167 @@ def build(module):
...
 
       # CHECK: comb.icmp uge %[[CONST]], %[[CONST]]
-      comb.GeUOp.create(lhs=const.result, rhs=const.result)
+      comb.GeUOp.create(const.result, const.result)
       geu = comb.GeUOp.create()
       connect(geu.lhs, const.result)
       connect(geu.rhs, const.result)
 
+      # CHECK: comb.add %[[CONST]], %[[CONST]]
+      comb.AddOp.create(const.result, const.result)
```
Comment (teqdruid) [inline integration_test/Bindings/Python/dialects/comb.py:161]: One of these tests should test >2 operands.

File: lib/Bindings/Python/circt/dialects/_comb_ops_ext.py [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
 
-from mlir.ir import IntegerAttr, IntegerType
+from mlir.ir import IntegerAttr, IntegerType, Type
 
 from functools import reduce
 
 
+def infer_result_type(operands):
+  types = list(map(lambda arg: get_value(arg).type, operands))
+  if not types:
```
Comment (teqdruid) [inline lib/Bindings/Python/circt/dialects/_comb_ops_ext.py:11]: Should this be `len(types) == 0`? Even if `not` does the same, I think the len is clearer.

File: lib/Bindings/Python/circt/dialects/_comb_ops_ext.py [hunk 3]
```diff
@@ -49,17 +60,45 @@ def BinaryOp(base):
...
+        if not lhs and not rhs:
           raise TypeError("result type must be specified")
-        all_equal = reduce(lambda t1, t2: t1 == t2, types)
-        if not all_equal:
-          raise TypeError(
-              f"expected same input port types, but received {types}")
-        result_type = types[0]
-      return BinaryOpBuilder(cls, result_type, kwargs)
+        rhs = lhs
+        lhs = result_type
```
Comment (teqdruid) [inline lib/Bindings/Python/circt/dialects/_comb_ops_ext.py:68]: `result_type` is inaccurately named if used this way. Maybe put it at the end and require it to be a keyword argument?

**Review by teqdruid — APPROVED** [review]

File: lib/Bindings/Python/circt/dialects/_comb_ops_ext.py (hunk 2 above)
Comment (mikeurbach) [reply lib/Bindings/Python/circt/dialects/_comb_ops_ext.py:11]: I think google style recommends `if not types`: https://google.github.io/styleguide/pyguide.html#2144-decision. But if it's more sane for an MLIR compiler developer, I can use `len(types) == 0`.

File: lib/Bindings/Python/circt/dialects/_comb_ops_ext.py (hunk 3 above)
Comment (mikeurbach) [reply lib/Bindings/Python/circt/dialects/_comb_ops_ext.py:68]: Yeah, I think that is a much better way.

File: lib/Bindings/Python/circt/dialects/_comb_ops_ext.py (hunk 2 above)
Comment (teqdruid) [reply lib/Bindings/Python/circt/dialects/_comb_ops_ext.py:11]: I'm OK either way. It just confuses me every time, but that's just a guy who doesn't really know python.

**Pull request merged**